src/framework/components/rigid-body/system.js

```diff
@@ -552,6 +600,275 @@ class RigidBodyComponentSystem extends ComponentSystem {
         return results;
     }
 
+    /**
+     * Perform a collision check on the world and return all entities the box hits.
+     * It returns an array of {@link HitResult}. If no hits are
+     * detected, the returned array will be of length 0.
+     *
+     * @param {Vec3} halfExtents - The half-extents of the box in the x, y and z axes. Cast distance will be added to Z-Axis.
+     * @param {Vec3} start - The world space point where the box starts.
+     * @param {Vec3} end - The world space point where the box ends.
+     *
+     * @returns {HitResult[]} An array of boxCast hit results (0 length if there were no hits).
+     */
+    boxCast(halfExtents, start, end) {
+        // Sweeping
+        halfExtents.z += start.distance(end) / 2;
+
+        // Find rotation
+        shapecastRotationMatrix.setLookAt(start, end, Vec3.UP);
+        shapecastRotation.setFromMat4(shapecastRotationMatrix);
+
+        // Transform start vector to make it beween initial start and end.
+        start.add(end).divScalar(2);
```

Comment:
Please don't change input vectors, otherwise vectors are messed up and everyone is scratching their head why the data in some other place changed :thought_balloon: :thinking: 

If I understand it right, it should be something like

1) On top of file a temp vector: `const startEndCenter = new Vec3();`
2) Instead of `start.add(end).divScalar(2);`: `startEndCenter.lerp(start, end, 0.5);`
3) Replace `start` with `startEndCenter` in `this._shapeTest` line

Reply:
Alright! That's updated